specs/anchored-region/anchored-region.md

```diff
@@ -0,0 +1,285 @@
+# Anchored region
+
+## Overview
+
+An *anchored region* is a container component which enables authors to create layouts where the contents of the anchored region can be positioned relative to another "anchor" element.  Additionally, the *anchored region* can react to the available space between the anchor and a parent ["viewport"](https://developer.mozilla.org/en-US/docs/Glossary/viewport) element such that the region is placed on the side of the anchor with the most available space, or even resize itself based on that space.
+
+### Background
+
+This component is inspired by the ["Viewport positioner"](https://github.com/microsoft/fast-dna/tree/master/packages/fast-components-react-base/src/viewport-positioner)  component in the React component set.  It is used as a building block in other components to enable responsive flyouts, or positionable/scaling menus in the [select](https://github.com/microsoft/fast-dna/tree/master/packages/fast-components-react-base/src/select) component. 
+
+A primary goal of the component was to enable authors to create responsive layouts without resorting to expensive [getBoundingClientRect()](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) calls. It instead depends on the more performant [IntersectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver) and [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) interfaces.
+
+For a more in-depth understanding of how this component works under the covers please refer to the [intersection observer api](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API). 
+
+### Use Cases
+
+It is envisioned that this component would be used as a building block for other components in this library (select, flyout, tooltip, etc) as well as being available for standalone use in responsive layouts.
+
+### Features
+
+- **Relative positioning:** Users can use it to position an element relative to another another element directly, like enabling a menu to open above or below a trigger button.
```

Comment:
It could be the language, or the way it's being described but technically in CSS I can already position something relative to another element. The value in the component here seems to be that it will ensure it doesn't render outside the viewport in the event of collision.

Reply:
The basic relative positioning has a convenience factor in that if you have a select component that responds to layout changes you could choose to turn off that functionality (ie. always open up) without having to have a different layout mechanism to support that.  But more importantly it allows the region to be anchored dynamically to any other element within its viewport regardless of whether they are relative to each other in the dom.

Reply:
Can you add that nuance to the document? I think it's important context.

Reply:
Added "Additonally, the same anchored region can change which element it is anchored to dynamically.  For example a single tooltip instance in a page could be positioned next to any other element on the page by switching the anchor property of the anchored region that contains it."